addOns/grpc/src/test/java/org/zaproxy/addon/grpc/internal/ProtoBufMessageDecoderUnitTest.java

```diff
@@ -57,9 +56,9 @@ void shouldDecodeWithSimpleValidInput() {
         String inputString =
                 "AAAAADEKC2pvaG4gTWlsbGVyEB4aIDEyMzQgTWFpbiBTdC4gQW55dG93biwgVVNBIDEyMzQ1";
         String expectedOutput =
-                "1:2::john Miller\n2:0::30\n3:2::1234 Main St. Anytown, USA 12345\n";
+                "1:2::\"john Miller\"\n2:0::30\n3:2::\"1234 Main St. Anytown, USA 12345\"\n";
         // Example valid input byte array
-        byte[] validInput = Base64.getDecoder().decode(inputString);
+        byte[] validInput = org.apache.commons.codec.binary.Base64.decodeBase64(inputString);
```

Comment:
Why use commons vs core?

Reply:
Throughout this PR

Reply:
This suggests that you're probably including an extra line feed in the value being decoded:
https://stackoverflow.com/questions/35292836/input-byte-array-has-incorrect-ending-byte-at-40

This answer:
https://stackoverflow.com/a/44708711/7718222
suggests simply trimming the input string to ensure there's no whitespace.

Reply:
i also tried to find out why it is breaking, and found this trimming solution but it is still giving  same error

Reply:
Hmmm.... I'll do more reading. Leave it for now

Reply:
See if `Base64.getUrlDecoder().decode behaves?
- https://stackoverflow.com/a/45740370/7718222
- https://stackoverflow.com/a/49009467/7718222
- https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html#url

Reply:
getUrlDecoder().decode and getMimeDecoder().decode  is giving same error

Reply:
currently i am trying for this input "AAAAABoKGEhlbGxvISBCbGpnbGthc2pnanNsZGpsZw==gAAAACBncnBjLXN0YXR1czowDQpncnBjLW1lc3NhZ2U6T0sNCg=="

Reply:
Why are there two sets of padding in a single string?

Reply:
don't know, i got this response from a sample app

Reply:
The first part seems to be a message `Hello! Bljglkasjgjsldjlg`
The second part seems to be the server response. I know nothing about the protocol so maybe I'm wrong but that's how it **seems** to me.
```
grpc-status:0
grpc-message:OK
```

Maybe they're meant to be together? But it seems strange that they're encoded separately. If the two values didn't happen to have padding you wouldn't know where the first one ended and the second started (unless there's a length identifier embedded?).

Is that how apache commons decodes it?

(Sorry don't have a dev box handy right now.)

Reply:
yes you are right, server is sending request status + data , i need to somehow split the encoded data to get data separately

Reply:
Maybe the server is returning something it shouldn't. Does Google have some sort of validator?

Reply:
actually server will send different status code to client, but i wonder why it is in body , why not some kind of header

Reply:
I see you've implemented a change for this. Are you sure there will always be padding (equals sign) in cases where this happens?